### PR TITLE
[WIP] Oracle Cloud Support

### PR DESCRIPTION
**Pull request by randomvariable**
### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
Added support for provisioning TKG on Oracle Cloud. Refer to product documentation for more information.
```

### CODE REVIEW

**Comment by HanFa** [comment]
Added supplementary documentations at https://github.com/vmware-tanzu/tanzu-framework/pull/3600 for this PR.

File: tkg/client/client.go [hunk 1]
```diff
@@ -74,12 +74,15 @@ type InitRegionOptions struct {
...
 	Edition                      string
+	AdditionalManifests          string
+	AdditionalManifestsNamespace string
 	Annotations                  map[string]string
 	Labels                       map[string]string
 	FeatureFlags                 map[string]string
 	LaunchUI                     bool
 	CeipOptIn                    bool
 	UseExistingCluster           bool
+	BootstrapClusterOnly         bool
```
Comment (randomvariable) [inline tkg/client/client.go:85]: Need to drop this code.

Comment (randomvariable) [inline tkg/client/client.go:85]: This needs to be moved to a separate PR and is dependent on Cluster API 1.3

File: apis/run/v1alpha3/tanzukubernetesrelease_types.go [hunk 2]
```diff
@@ -40,12 +40,25 @@ type TanzuKubernetesReleaseSpec struct {
...
 	BootstrapPackages []corev1.LocalObjectReference `json:"bootstrapPackages,omitempty"`
 }
 
+// OSImageReference specifies the details about the osimage reference in the TanzuKubernetesRelease objects.
+type OSImageReference struct {
+	corev1.LocalObjectReference `json:",inline"`
+
+	// The URL of public image endpoint where the image will be imported from
+	// +optional
+	ImageURL *string `json:"image_url,omitempty"`
```
Comment (imikushin) [inline apis/run/v1alpha3/tanzukubernetesrelease_types.go:55]: **nit**: json field should be named `imageURL`: in line with generally accepted style in Kubernetes APIs.

File: cmd/cli/plugin/managementcluster/create.go [hunk 3]
```diff
@@ -135,6 +141,9 @@ func init() {
 	createCmd.Flags().BoolVar(&iro.forceConfigUpdate, "force-config-update", false, "Force an update of all configuration files in ${HOME}/.config/tanzu/tkg/bom and ${HOME}/.tanzu/tkg/compatibility")
 
 	createCmd.Flags().SetNormalizeFunc(aliasNormalizeFunc)
+
+	createCmd.Flags().StringVarP(&iro.additionalManifests, "additional-manifests", "", "", "Additional manifests to be applied to the bootstrap cluster")
+	createCmd.Flags().StringVarP(&iro.additionalManifestsNamespace, "additional-manifests-namespace", "", "", "Namespace of the additional manifest")
```
Comment (randomvariable) [inline cmd/cli/plugin/managementcluster/create.go:146]:
```suggestion
	createCmd.Flags().StringVarP(&iro.additionalManifestsNamespace, "additional-manifests-namespace", "", "", "Namespace of the additional manifest")
```

For the purposes of management-cluster plugin, would it ever anything but `tkg-system` since the management cluster MUST be created in that namespace.

Comment (HanFa) [reply cmd/cli/plugin/managementcluster/create.go:146]: yep, statement is true. i am thinking renaming the flag to `--additional-tkg-system-manifests` and drop the other flag for specifying the NS

**Comment by randomvariable** [comment]
Superseded by multiple PRs.
